runtime/common/src/weights/xcmp_queue.rs

```diff
@@ -70,4 +70,28 @@ impl<T: frame_system::Config> WeightInfo for HydraWeight<T> {
 			.saturating_add(T::DbWeight::get().reads(1 as u64))
 			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
+	// Storage: XcmpQueue QueueConfig (r:1 w:0)
+	// Proof Skipped: XcmpQueue QueueConfig (max_values: Some(1), max_size: None, mode: Measured)
+	// Storage: XcmpQueue DeferredXcmMessages (r:1 w:1)
+	// Proof Skipped: XcmpQueue DeferredXcmMessages (max_values: None, max_size: None, mode: Measured)
+	// Storage: XcmpQueue CounterForOverweight (r:1 w:1)
+	// Proof: XcmpQueue CounterForOverweight (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	// Storage: XcmpQueue OverweightCount (r:1 w:1)
+	// Proof Skipped: XcmpQueue OverweightCount (max_values: Some(1), max_size: None, mode: Measured)
+	// Storage: XcmpQueue Overweight (r:100 w:100)
+	// Proof Skipped: XcmpQueue Overweight (max_values: None, max_size: None, mode: Measured)
+	fn service_deferred() -> Weight {
+		// Minimum execution time: 90_934_683 nanoseconds.
+		Weight::from_ref_time(91_491_151_000 as u64)
+			.saturating_add(T::DbWeight::get().reads(104 as u64))
+			.saturating_add(T::DbWeight::get().writes(103 as u64))
```

Comment:
100 reads/writes? just looks suspicious.

Reply:
Yeah, the worst case is quite terrible: it happens when all messages are overweight and involves a write for every message.

Reply:
what impact this will have on the cost of transfer ?

Reply:
this has impact on production runtime even without integration enabled

Reply:
i guess this function is called by the normal xcm inherent as well, do we account for this extra weight consumed?

Reply:
The XCMP queue actually does not account for the weight of storing overweight messages.
So we're doing no worse than the previous implementation, but it's not great, yeah.

Reply:
> what impact this will have on the cost of transfer ?

I don't think it will have any? These XCM processing costs are eaten by the chain